kclvm/tools/src/LSP/src/find_refs.rs

```diff
@@ -1,23 +1,64 @@
 use crate::from_lsp::kcl_pos;
-use crate::goto_def::goto_definition;
+use crate::goto_def::{find_def, goto_definition};
+use crate::to_lsp::lsp_location;
 use crate::util::{parse_param_and_compile, Param};
 use anyhow;
+use kclvm_ast::ast::{Program, Stmt};
+use kclvm_error::Position as KCLPos;
+use kclvm_sema::resolver::scope::ProgramScope;
 use lsp_types::{Location, Url};
 use parking_lot::RwLock;
 use ra_ap_vfs::Vfs;
 use std::collections::HashMap;
 use std::sync::Arc;
 
 pub(crate) fn find_refs<F: Fn(String) -> Result<(), anyhow::Error>>(
+    program: &Program,
+    kcl_pos: &KCLPos,
+    prog_scope: &ProgramScope,
+    word_index_map: HashMap<Url, HashMap<String, Vec<Location>>>,
+    vfs: Option<Arc<RwLock<Vfs>>>,
+    logger: F,
+) -> Result<Vec<Location>, String> {
+    // find the definition at the position
+    let def = match program.pos_to_stmt(kcl_pos) {
+        Some(node) => match node.node {
+            Stmt::Import(_) => None,
+            _ => find_def(node.clone(), kcl_pos, prog_scope),
+        },
+        None => None,
+    };
+    if def.is_none() {
+        return Err(String::from(
+            "Definition item not found, result in no reference",
+        ));
+    }
+    let def = def.unwrap();
+    if def.get_positions().len() != 1 {
```

Comment:
The description `more than one definitions` means the length of positions condition must be the form `def.get_positions().len() > 1`?

Reply:
fixed